rusk/src/lib/http/chain.rs

```diff
@@ -253,4 +263,47 @@ impl RuskNode {
 
         Ok(ResponseData::new(serde_json::to_value(stats)?))
     }
+
+    async fn simulate_tx(&self, tx: &str) -> anyhow::Result<ResponseData> {
+        let tx = hex::decode(tx).map_err(|e| {
+            anyhow::anyhow!("Failed to decode transaction: {e}")
+        })?;
+        let tx = ProtocolTransaction::from_slice(&tx)
+            .map_err(|e| anyhow::anyhow!("Invalid Data: {e:?}"))?;
+        let vm_handler = self.inner().vm_handler();
+        let vm_handler = vm_handler.read().await;
+        if tx.gas_limit() > vm_handler.get_block_gas_limit() {
+            return Err(anyhow::anyhow!("Gas limit is too high."));
+        }
+        let tip = load_tip(&self.db())
+            .await
+            .map_err(|e| anyhow::anyhow!("Failed to load the tip: {e}"))?;
+        let Some(tip) = tip else {
+            return Err(anyhow::anyhow!("Could not find the tip"));
+        };
```

Comment:
The statement `let else` is very confusing to my

Can you use the ok_or?

```suggestion
        let tip = load_tip(&self.db())
            .await
            .map_err(|e| anyhow::anyhow!("Failed to load the tip: {e}"))?
            .ok_or(|e| anyhow::anyhow!("Could not find the tip"))?;
```